ibis/pandas/execution/window.py

```diff
@@ -130,24 +132,64 @@ def get_aggcontext_window(
     return aggcontext
 
 
+def trim_with_timecontext(data, timecontext):
+    try:
+        begin, end = map(pd.to_datetime, timecontext)
+    except ValueError:
+        raise com.IbisError('Cannot resolve timecontext for window\n{}.')
+    else:
+        df = data.reset_index()
+        time_col = 'time'
+        subset = df.loc[df[time_col].between(begin, end)]
+        name = data.name
+        non_target_columns = list(subset.columns.difference([name]))
```

Comment:
Can you explain what is non_target_columns and why do we need it? I wonder why this is simply a filtering.

Reply:
explained in inline doc, these are indexes and we need them to reset the correct index back to the Series